### DISCUSSION TITLE
Copilot authentication failure

### DISCUSSION THREAD

**Discussion by alexturnbull** [community/community#118416] [Copilot Conversations]
### VS code authentication 

Bug

My copilot was working fine on both my ubuntu pc in vscode for a few weeks and then suddenly was not able to connect to the server. I've tried to access it from my windows PC and the same thing happens. 
1. it prompts me to login and a blank screen opens on my browser and that eventually fails  
2. it prompts me to try locally and that opens a window in my browser saying you are logged in and it eventually fails
3. i try device code and that fails 
not sure why it suddenly stopped working but if anyone else has had this issue would be great to know how you got passed it? 

Thanks

**Comment by LunaCheung** [comment]
This error indicates that you do not have a GitHub Copilot subscription, or that an error occurred while connecting to the GitHub API to request a token to utilize GitHub Copilot.

To request another token from api.github.com, sign in and out of GitHub Copilot in your IDE. After you log out, GitHub Copilot will prompt you to sign back in. Also check your sub if it is still in period.

**Reply by alexturnbull** [reply]
Hi YJ, I've logged in and out a few times also trying different browsers and restarting the PC's, I've uninstalled and reinstalled copilot including older versions, I've also done fresh installs of vs code. I've checked my subscription I'm still on the free trial with the first payment due on the 19th of this month (april) i intend to keep using it if I can get it working but I'm running out of ideas. like i say it was working and just seemed to stop last week, do you know of any other reason for it not authenticating? or of i way i could do this manually? thanks, Alex

**Reply by LunaCheung** [reply]
1. I think it also could be the cache leftover in your system. Try clean your browser cache up or use another browser. 
2. Also, i guess VS Code has a lot of addons, there are some corruppting maybe? Try stop/uninstall/update them and test. 
3. Type "Developer: Toggle Developer Tools" in command palette and check some error logs. 
4. If you are using proxy, make sure proxy are setted correctly. Since I used jetbrains, in company, they use proxy and u have to set proxy inside the IDE. 
5. Check "GitHub OAuth Applications" in your account settings, see if VSCode is added. 
6. Close Firewall

If all those things not working, i have no idea maybe the addon is having bug. Then you need to write a ticket to admin. 🙂
